Fix tests for site area

roxiciobotea210 opened the pull request:
The initial version of tests were using some default entities and once those were modified in the first run, the tests were failing.

roxiciobotea210 commented:
@LucasBrazi06 All `org` tests are passing at this point.

jerome-benoit commented on test/api/SiteAreaOrgTest.ts:
```diff
@@ -308,79 +379,83 @@ describe('Site Area tests', function() {
-          testData.siteAreaContext.getSiteArea().id, [testData.chargingStationContext.getChargingStation().id]);
-        expect(response.status).to.equal(StatusCodes.FORBIDDEN);
+          testData.createdSiteAreas[testData.createdSiteAreas.length - 1].id, [chargingStations.data.result[0].id]);
+        expect(response.status).to.equal(403);
```
Please use `StatusCodes.` instead.

jerome-benoit commented on test/api/SiteAreaOrgTest.ts:
```diff
@@ -308,79 +379,83 @@ describe('Site Area tests', function() {
-      // Try to assign ChargingStation to SiteArea
+        // Try to assign ChargingStation to SiteArea
+        const chargingStations = await testData.userService.chargingStationApi.readAll({});
+        expect(chargingStations.status).to.equal(200);
```
Please use StatusCodes. instead.

jerome-benoit commented on test/api/SiteAreaOrgTest.ts:
```diff
@@ -308,79 +379,83 @@ describe('Site Area tests', function() {
-      // Try to remove ChargingStation from SiteArea
+        // Try to remove ChargingStation from SiteArea
+        const chargingStations = await testData.userService.chargingStationApi.readAll({});
+        expect(chargingStations.status).to.equal(200);
```
Idem

jerome-benoit commented on test/api/SiteAreaOrgTest.ts:
```diff
@@ -308,79 +379,83 @@ describe('Site Area tests', function() {
-          testData.siteAreaContext.getSiteArea().id, [testData.chargingStationContext.getChargingStation().id]);
-        expect(response.status).to.equal(StatusCodes.FORBIDDEN);
+          testData.createdSiteAreas[testData.createdSiteAreas.length - 1].id, [chargingStations.data.result[0].id]);
+        expect(response.status).to.equal(403);
```
Idem

jerome-benoit commented on test/api/SiteAreaOrgTest.ts:
```diff
@@ -308,79 +379,83 @@ describe('Site Area tests', function() {
-          testData.siteAreaContext.getSiteArea().id, [testData.chargingStationContext.getChargingStation().id]);
-        expect(response.status).to.equal(StatusCodes.FORBIDDEN);
+          testData.createdSiteAreas[testData.createdSiteAreas.length - 2].id, [chargingStations.data.result[0].id]);
+        expect(response.status).to.equal(200);
```
Idem

jerome-benoit reviewed (changes requested):
Please use StatusCodes instead of raw HTTP error code for the sake of code readability. 

I have a concern regarding the use of the createdSiteAreas array, it really makes the code hard to understand: what is the difference between `createdSiteAreas[testData.createdSiteAreas.length - 1]` and `createdSiteAreas[testData.createdSiteAreas.length - 2]`?
Is the use of the array really needed? It is possible to have explicit siteAreas name instead coming from testData? 

Thanks.

LucasBrazi06 commented:
@roxiciobotea210 To move to To be reviewed pipeline?

roxiciobotea210 commented:
> Please use StatusCodes instead of raw HTTP error code for the sake of code readability.
> 
> I have a concern regarding the use of the createdSiteAreas array, it really makes the code hard to understand: what is the difference between `createdSiteAreas[testData.createdSiteAreas.length - 1]` and `createdSiteAreas[testData.createdSiteAreas.length - 2]`?
> Is the use of the array really needed? It is possible to have explicit siteAreas name instead coming from testData?
> 
> Thanks.

I replaced explicit codes with `StatusCodes`, I rewrote them at merge by mistake.
Also, I added explicit names in `testData` for the`siteAreas` to avoid the confusion with "-1", "-2", you are right, it was hard to understand.

roxiciobotea210 commented:
> @roxiciobotea210 To move to To be reviewed pipeline?

Done :)

jerome-benoit commented on test/api/SiteAreaOrgTest.ts:
```diff
@@ -285,7 +357,7 @@ describe('Site Area tests', function() {
           false
         );
-        expect(response.status).to.equal(StatusCodes.FORBIDDEN);
+        expect(response.status).to.equal(500);
```
Missed `StatusCodes` usage ;)

roxiciobotea210 replied:
Fixed :)

jerome-benoit reviewed (approved):
Small missing bits.
